pkg/ast/spl/tests/splParser_test.go

```diff
@@ -7792,3 +7792,62 @@ func Test_Format_cmd_Incomplete_RowCol_Options(t *testing.T) {
 	_, err := spl.Parse("", []byte(query))
 	assert.NotNil(t, err)
 }
+
+func Test_EventCount_Defaults(t *testing.T) {
+	query := `* | eventcount`
+	performEventCountTest(t, query, "*", true, false, true)
+}
+
+func Test_EventCount_IndexSpecified(t *testing.T) {
+	query := `* | eventcount index=my_index`
+	performEventCountTest(t, query, "my_index", true, false, true)
+}
+
+func Test_EventCount_SummarizeFalse(t *testing.T) {
+	query := `* | eventcount summarize=false`
+	performEventCountTest(t, query, "*", false, false, true)
+}
+
+func Test_EventCount_ReportSizeTrue(t *testing.T) {
+	query := `* | eventcount report_size=true`
+	performEventCountTest(t, query, "*", true, true, true)
+}
+
+func Test_EventCount_ListVixFalse(t *testing.T) {
+	query := `* | eventcount list_vix=false`
+	performEventCountTest(t, query, "*", true, false, false)
+}
+
+func Test_EventCount_Combination1(t *testing.T) {
+	query := `* | eventcount index=my_index summarize=false report_size=true`
+	performEventCountTest(t, query, "my_index", false, true, true)
+}
+
+func Test_EventCount_Combination2(t *testing.T) {
+	query := `* | eventcount index=my_index summarize=true report_size=false list_vix=false`
+	performEventCountTest(t, query, "my_index", true, false, false)
+}
```

Comment:
You should also add a unit test where the optional parameters are in a different order. (I'm pretty sure your existing parsing solution will handle this correctly)

Reply:
Added more testcases for parser, and also updated the parser to handle multiple index names in the query like 

`eventcount summarize=false index=_audit index=main` and added test cases for this as well in multiple combinations